## Problem Statement 01

A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions.

# Data Description
### alpha =0.05(95% Confidence)
### Is Y1 and Y2 are normal
### HO = Y1 and Y2 are normal
### H1 = Y1 and Y2 are not normal


In [11]:
import pandas as pd
import numpy as np
from scipy import stats

In [12]:
df  = pd.read_csv('Cutlets.csv')

In [13]:
df.sample(5)

Unit A  Unit B
19  7.3495  6.7478
30  6.7794  7.0992
28  6.9431  6.3346
24  6.9952  6.7594
11  7.5169  7.4059

In [14]:
# Normality Test For Unit A
from scipy.stats import shapirostat, p = shapiro(df["Unit A"])
print(stat,p)

0.9649459719657898 0.31998491287231445


In [15]:
# Normality Test For Unit B
from scipy.stats import shapiro
stat, p = shapiro(df["Unit B"])
print(stat,p)

0.9727305769920349 0.5225146412849426


##### P value for unit A == 0.3199> alpha
##### P value for unit B == 0.522> alpha
### Ho is accepted. Y1 and Y2 are normal

## The Data is normally distributed now we perform T-test
#### HO == Y1 and Y2 Are Equal(There is no significance difference between     diameter of the Cutlets)
#### H1 == Y1 and Y2 are not Equal((There is a significance difference between diameter of the Cutlets)

In [16]:
#T-test
stats.ttest_rel(df["Unit A"], df["Unit B"])

Ttest_relResult(statistic=0.7536787225614316, pvalue=0.4562300768038412)

### P value of the Paired T Test is == 0.45 > alpha
#### Thus H0 is accepted.
#### Mean of both Y1 and Y2 are equal


## Problem Statement 02
####   A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch.Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level.


## Data Description
##### alpha ==0.05(95% Confidence)
##### H0 = Y1,Y2,Y3 and Y4 are normal
##### H1 = Y1,Y2,Y3 and Y4 are not normal


In [18]:
df2 =pd.read_csv('LabTAT.csv')

In [19]:
df2.sample(5)

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
68        201.27        148.70        215.17        181.76
40        175.50        186.49        189.61        157.02
95        171.64        188.49        202.95        167.62
5         179.45        197.00        181.51        161.12
94        168.62        159.92        204.28        172.38

In [25]:
#Normality Test For Y1 
stat, p = shapiro(df2["Laboratory 1"])
print(stat,p)

0.990183413028717 0.5507769584655762


In [26]:
#Normality Test For Y2
stat, p = shapiro(df2["Laboratory 2"])
print(stat,p)

0.9936320185661316 0.8637352585792542


In [27]:
#Normality Test For Y3
stat, p = shapiro(df2["Laboratory 3"])
print(stat,p)

0.988633930683136 0.42045730352401733


In [28]:
#Normality Test For Y4
stat, p = shapiro(df2["Laboratory 4"])
print(stat,p)

0.9913760423660278 0.6619619131088257


### Y1,Y2,Y3 and Y4 are >alpha
## Ho is accepted.

## Variance
#### H0 == Variance of all 4 laboratories are the same
#### H1 == Variance of all 4 laboratories are the not same

In [28]:
stats.levene(df2["Laboratory 1"], df2["Laboratory 2"], df2["Laboratory 3"], df2["Laboratory 4"])

LeveneResult(statistic=2.599642500418024, pvalue=0.05161343808309816)

#### P Value of Variance test is == 0.051 >alpha
#### H0 is Accepted

### H0 is accepted so we can perform one way anova test
#### H0 == Mean TAT for 4 laboratories equal (There is no significance difference between TAT of the laboratories)
#### H1 == Mean TAT for 4 laboratories not equal (There is a significance difference between TAT of the laboratories)

In [29]:
stats.stats.f_oneway(df2["Laboratory 1"], df2["Laboratory 2"], df2["Laboratory 3"], df2["Laboratory 4"])

C:\Users\PRIYANKA\AppData\Local\Temp\ipykernel_8864\2972643213.py:1: DeprecationWarning: Please use `f_oneway` from the `scipy.stats` namespace, the `scipy.stats.stats` namespace is deprecated.
  stats.stats.f_oneway(df2["Laboratory 1"], df2["Laboratory 2"], df2["Laboratory 3"], df2["Laboratory 4"])


F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

## P value of the One way Anova test is == 2e-16 < alpha

####  H1 is accepted.
#### Mean TAT for 4 laboratories not equal (There is a significance difference between TAT of the laboratories)

## Problem Statement 03
##### Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions.


## Data Description
#### alpha = 0.05(95% Confidence)
##### Since there are more than 2 variable we will perform Chi-Square test

In [32]:
df3 = pd.read_csv('BuyerRatio.csv')

In [49]:
df3.head()

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [39]:
# Chi-Square Test
#H0 == The male-female buyer rations are similar across regions
#H1 == The male-female buyer rations are not similar across regions

stats.chi2_contingency([df3["East"], df3["West"], df3["North"], df3["South"]])


(1.5959455386610577,
 0.6603094907091882,
 3,
 array([[  42.76531299,  442.23468701],
        [ 146.81287862, 1518.18712138],
        [ 131.11756787, 1355.88243213],
        [  72.30424052,  747.69575948]]))

#### P-value Of Chi-Square Test  = 0.66030> alpha
###  HO is accepted.
### The male-female buyer rations are similar across regions

## Problem Statement 04

##### TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing.  The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences

In [42]:
# Data Discription
#alpha = 0.05(95% Confidence Interval).
#Since There are more than 2 variable so we can perform Chi-Square test.
df4 = pd.read_csv('Costomer+OrderForm.csv')

In [49]:
df4.head(5)

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

## Chi-Square Test
#### H0 == The defective % does not varies by centre
#### H1 == The defective % does varies by centre

In [53]:
stats.chi2_contingency([df4['Phillippines'].value_counts(), df4['Indonesia'].value_counts(), df4['Malta'].value_counts(), df4['India'].value_counts()])

(3.8589606858203545,
 0.2771020991233144,
 3,
 array([[271.75,  28.25],
        [271.75,  28.25],
        [271.75,  28.25],
        [271.75,  28.25]]))

## P value of Chi-Square test == 0.277 > alpha
### HO is accepted.
### defective % does not varies significantly by centres